functions/postSurveyComplete.protected.ts

```diff
@@ -0,0 +1,142 @@
+/* eslint-disable @typescript-eslint/camelcase */
+/* eslint-disable import/no-dynamic-require */
+/* eslint-disable global-require */
+import {
+  Context,
+  ServerlessCallback,
+  ServerlessFunctionSignature,
+} from '@twilio-labs/serverless-runtime-types/types';
+import axios from 'axios';
+// eslint-disable-next-line prettier/prettier
+import type { TaskInstance } from 'twilio/lib/rest/taskrouter/v1/workspace/task';
+import type { BuildSurveyInsightsData, OneToManyConfigSpecs } from './helpers/insightsService.private';
+import type { BuildDataObject, PostSurveyData } from './helpers/hrmService.private';
+
+export type BotMemory = {
+  memory: {
+    twilio: { collected_data: { collect_survey: { [question: string]: string | number } } };
+  };
+};
+
+type PostSurveyBody = {
+  contactTaskId: string;
+  taskId: string;
+  data: PostSurveyData;
+};
+
+export interface Event {
+  Channel: string;
+  CurrentTask: string;
+  Memory: string;
+  UserIdentifier: string;
+}
+
+type EnvVars = {
+  TWILIO_WORKSPACE_SID: string;
+  HRM_STATIC_KEY: string;
+};
+
+const saveSurveyInInsights = async (postSurveyConfigJson: OneToManyConfigSpecs, memory: BotMemory, surveyTask: TaskInstance) => {
+  const handlerPath = Runtime.getFunctions()['helpers/insightsService'].path;
+  const buildSurveyInsightsData = require(handlerPath)
+    .buildSurveyInsightsData as BuildSurveyInsightsData;
+
+  const taskAttributes = JSON.parse(surveyTask.attributes);
+  const finalAttributes = buildSurveyInsightsData(postSurveyConfigJson)(taskAttributes, memory);
+  console.log('finalAttributes: ', JSON.stringify(finalAttributes));
+
+  await surveyTask.update({ attributes: JSON.stringify(finalAttributes) });
+};
+
+const saveSurveyInHRM = async (postSurveyConfigJson: OneToManyConfigSpecs, memory: BotMemory, surveyTask: TaskInstance, hrmBaseUrl: string, hrmStaticKey: string) => {
+  const handlerPath = Runtime.getFunctions()['helpers/hrmService'].path;
+  const buildDataObject = require(handlerPath)
+    .buildDataObject as BuildDataObject;
+
+  const taskAttributes = JSON.parse(surveyTask.attributes);
+
+  const data = buildDataObject(postSurveyConfigJson, memory);
+
+  const body: PostSurveyBody = {
+    contactTaskId: taskAttributes.contactTaskId,
+    taskId: surveyTask.sid,
+    data
+  };
+
+  await axios({
```

Comment:
Please add a comment explaining why we're using axios, especially since most of our code does not.

Reply:
Can you add this comment?

Reply:
Never mind... I see that it's at the top.